### DISCUSSION TITLE
Option to show a pop confirmation before route change

### DISCUSSION THREAD

**Discussion by AkashGoenka** [vercel/next.js#27242] [Ideas]
### Describe the feature you'd like to request

I had come across a use case where we had to warn user before they leave the page to another URL or in App Navigation. It would be great to have a feature to perform some logic before route change is initiated.

### Describe the solution you'd like

A router event like `beforeRouteChange` that triggers a callback function where additional business logic can be executed and the routing can be halted/continued on the basis of a boolean returned by the callback.

Another option would be something similar to https://reactrouter.com/core/api/Prompt 

### Describe alternatives you've considered

I've tried to use window.onbeforeunloadevent, but this does not work in all scenarios (especially in app navigation). It is triggered mostly on closing the tab/browser window.

Another option was a workaround suggested in - https://github.com/vercel/next.js/issues/2476#issuecomment-563190607

**Comment by wadehammes** [comment]
You could build something similar using `router.events` -> https://nextjs.org/docs/api-reference/next/router#routerevents

**Comment by AkashGoenka** [comment]
https://github.com/vercel/next.js/issues/2476#issuecomment-563190607 makes use of https://nextjs.org/docs/api-reference/next/router#routerevents. But that seems more like a workaround (not the cleanest way). I want to implement some checks before the next framework starts to unmount the component.

**Reply by AkashGoenka** [reply]
Hey guys, just wanted to know if this is going to be considered in any of the future updates to next/router
